client/lib/feed-stream-store/feed-stream.js

```diff
@@ -488,22 +500,27 @@ export default class FeedStream {
 
 		const postById = this.postById;
 		forEach( this.pendingPostKeys, function( postKey ) {
-			postById.add( postKey.postId );
+			postById.add( keyToString( postKey ) );
 		} );
 
 		const mostRecentPostDate = moment( this.postKeys[ 0 ][ this.dateProperty ] );
 
-		if ( this.pendingDateAfter > mostRecentPostDate ) {
+		// if the first element in the current postKeys isn't also in the pending set,
+		// we have a possible gap
+		const firstKey = this.postKeys[ 0 ];
+		if (
+			! some( this.pendingPostKeys, pendingKey => keysAreEqual( pendingKey, firstKey ) ) &&
+			this.pendingDateAfter > mostRecentPostDate
```

Comment:
e.g. lets say there have been 10 posts w/ new comments.  all of them are already in the user's stream.  api request to get new convos returns 4 out of the 10.  how would we know that there are 6 more to fetch?